include/drivers/gpio.h

```diff
@@ -34,30 +34,34 @@ extern "C" {
  * @{
  */
 
+/** @cond INTERNAL_HIDDEN */
+#define GPIO_BASE_OFFSET(o) ((sizeof(gpio_dt_flags_t)*8) + (o))
```

Comment:
I'm not sure about the macro. If we need to extend `gpio_dt_flags_t` to 32-bits in the future it will be a hindrance not help. 

This could be useful if we decided that voltage flags are vendor specific. In that case the size of `gpio_dt_flags_t` could be made configurable via Kconfig option and we would support vendor specific DT flags optionally in bits 8-15. But at this point it looks like we shouldn't treat voltage flags as vendor specific.

Reply:
I found the macro to be self documenting with no run time cost.  Without this we have an offset of a magic 8 or 16 offset that once you know about gpio_dt_flags_t it is obvious but not so much before.  I will remove this but think it was clearer with the macro.